src/ChangesReporting/Output/GitlabOutputFormatter.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+/**
+ * CodeClimate Specification:
+ * - https://github.com/codeclimate/platform/blob/master/spec/analyzers/SPEC.md
+ */
+
+declare(strict_types=1);
+
+namespace Rector\ChangesReporting\Output;
+
+use Nette\Utils\Json;
+use Rector\ChangesReporting\Contract\Output\OutputFormatterInterface;
+use Rector\ValueObject\Configuration;
+use Rector\ValueObject\ProcessResult;
+
+final readonly class GitlabOutputFormatter implements OutputFormatterInterface
+{
+    /**
+     * @var string
+     */
+    public const NAME = 'gitlab';
+
+    private const ERROR_TYPE_ISSUE        = 'issue';
+    private const ERROR_CATEGORY_BUG_RISK = 'Bug Risk';
+    private const ERROR_CATEGORY_STYLE    = 'Style';
+    private const ERROR_SEVERITY_BLOCKER  = 'blocker';
+    private const ERROR_SEVERITY_MINOR    = 'minor';
+
+    public function getName(): string
+    {
+        return self::NAME;
+    }
+
+    public function report(ProcessResult $processResult, Configuration $configuration): void
+    {
+        $errorsJson = [
+            ...$this->appendSystemErrors($processResult, $configuration),
+            ...$this->appendFileDiffs($processResult, $configuration),
+        ];
+
+        $json = Json::encode($errorsJson, true);
+
+        echo $json . PHP_EOL;
+    }
+
+    /**
+     * @return array<array{
+     *      type: 'issue',
+     *      categories: array{'Bug Risk'},
+     *      severity: 'blocker',
+     *      description: string,
+     *      check_name: string,
+     *      location: array{
+     *          path: string,
+     *          lines: array{
+     *              begin: int,
+     *          },
+     *      },
+     *  }>
+     */
+    private function appendSystemErrors(ProcessResult $processResult, Configuration $configuration): array
+    {
+        $errorsJson = [];
+
+        foreach ($processResult->getSystemErrors() as $error) {
+            $filePath = $configuration->isReportingWithRealPath()
+                ? ($error->getAbsoluteFilePath() ?? '')
+                : ($error->getRelativeFilePath() ?? '')
+            ;
+
+            $fingerprint = md5($filePath . ';' . $error->getLine() . ';' . $error->getMessage());
```

Comment:
is md5 required for gitlab? we have `Filehasher::hash()` 

https://github.com/rectorphp/rector-src/blob/abaf47c828502d09a968590292aafe71d34e3e17/src/Util/FileHasher.php#L17-L20

Reply:
No, it's not required, I fixed it